pkg/scheduler/framework/v1alpha1/framework.go

```
@@ -49,44 +50,100 @@ const (
+			}
+			f.reservePlugins = append(f.reservePlugins, p)
+		} else {
+			return nil, fmt.Errorf("reserve plugin %v does not exist", r.Name)
 		}
+	}
 
-		if rp, ok := p.(ReservePlugin); ok {
-			f.reservePlugins = append(f.reservePlugins, rp)
+	for _, pb := range plugins.EnabledPlugins.PreBind {
+		if pg, ok := f.plugins[pb.Name]; ok {
+			p, ok := pg.(PrebindPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend prebind plugin", pb.Name)
+			}
+			f.prebindPlugins = append(f.prebindPlugins, p)
+		} else {
+			return nil, fmt.Errorf("prebind plugin %v does not exist", pb.Name)
 		}
-		if pp, ok := p.(PrebindPlugin); ok {
-			f.prebindPlugins = append(f.prebindPlugins, pp)
+	}
+
+	for _, ur := range plugins.EnabledPlugins.Unreserve {
+		if pg, ok := f.plugins[ur.Name]; ok {
+			p, ok := pg.(UnreservePlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend unreserve plugin", ur.Name)
+			}
+			f.unreservePlugins = append(f.unreservePlugins, p)
+		} else {
+			return nil, fmt.Errorf("unreserve plugin %v does not exist", ur.Name)
 		}
-		if up, ok := p.(UnreservePlugin); ok {
-			f.unreservePlugins = append(f.unreservePlugins, up)
+	}
+
+	for _, pr := range plugins.EnabledPlugins.Permit {
+		if pg, ok := f.plugins[pr.Name]; ok {
+			p, ok := pg.(PermitPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend permit plugin", pr.Name)
+			}
+			f.permitPlugins = append(f.permitPlugins, p)
+		} else {
+			return nil, fmt.Errorf("permit plugin %v does not exist", pr.Name)
 		}
-		if pr, ok := p.(PermitPlugin); ok {
-			f.permitPlugins = append(f.permitPlugins, pr)
+	}
+
+	for _, qs := range plugins.EnabledPlugins.QueueSort {
+		if pg, ok := f.plugins[qs.Name]; ok {
+			p, ok := pg.(QueueSortPlugin)
+			if !ok {
+				return nil, fmt.Errorf("plugin %v does not extend queue sort plugin", qs.Name)
+			}
+			f.queueSortPlugins = append(f.queueSortPlugins, p)
```

For now, we allow only one QueueSort plugin. We should throw an error if more than one is specified.

---

Updated. PTAL.
Thanks!!